### PR TITLE
[DOC] Fix deploy doc on main

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to #4401 

Changes proposed in this pull request:

- let github push to the documentation repo when building the doc on main as we already do for the doc of the release
- circle CI is now only there for hosting the doc of the PRs (that deploy still needs fixing).

### CODE REVIEW

**Comment by Remi-Gau** [comment]
The effect won't be visible until this is merged but I tested this on my forks.

this commit on my [main branch](https://github.com/Remi-Gau/nilearn/commit/a9a1d27834c433481faec6dc45ea502d5dd598de)

triggered this workflow: https://github.com/Remi-Gau/nilearn/actions/runs/9580817143

image

that pushed to my fork of the doc: https://github.com/Remi-Gau/nilearn/actions/runs/9580817143/job/26419321064#step:4:106

see commit here: https://github.com/Remi-Gau/nilearn.github.io/commit/30183edfcf61ea920966c4ea8083db5d40725ec3

and end result here: https://remi-gau.github.io/nilearn.github.io/dev/

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -20,10 +20,10 @@ on:
         branches:
         -   '*'
     schedule:
-    # Run every Monday at 8am UTC
+        # Run every Monday at 8am UTC
```
Comment (Remi-Gau) [inline .github/workflows/build-docs.yml:23]:
```suggestion
    # Run every Monday at 8am UTC
```

File: .github/workflows/build-docs.yml [hunk 2]
```diff
@@ -20,10 +20,10 @@ on:
...
     -   cron: 0 8 * * 1
 
-    # Allows you to run this workflow manually from the Actions tab
+        # Allows you to run this workflow manually from the Actions tab
```
Comment (Remi-Gau) [inline .github/workflows/build-docs.yml:26]:
```suggestion
    # Allows you to run this workflow manually from the Actions tab
```

**Comment by bthirion** [comment]
Can you clarify: where will the doc be found for all PRs ?
On https://remi-gau.github.io/nilearn.github.io/dev/ ?

**Comment by Remi-Gau** [comment]
> Can you clarify: where will the doc be found for all PRs ? On https://remi-gau.github.io/nilearn.github.io/dev/ ?

Oh no!!! The doc for each PR would still be deployed on circle-CI.

```mermaid
flowchart LR
    subgraph Circle-CI
    nilearn_CCI["`nilearn
    (host)`"]
    end
    subgraph GitHub
    nilearn_GH["`nilearn
    (build)`"]
    nilearn_doc["`nilearn.github.io
    (host)`"]
    end
    nilearn_GH --> nilearn_CCI

```

But for the doc building on main (the dev version of the doc) we were doing:

```mermaid
flowchart TD
    subgraph Circle-CI
    nilearn_CCI["`nilearn
    (host)`"]
    end
    subgraph GitHub
    nilearn_GH["`nilearn
    (build)`"]
    nilearn_doc["`nilearn.github.io
    (host)`"]
    end
    nilearn_GH --> nilearn_CCI
    nilearn_CCI --> nilearn_doc
```

After this PR it would be ("same" workflow as when we deploy the doc for one of our release)

```mermaid
flowchart TD
    subgraph Circle-CI
    nilearn_CCI["`nilearn
    (host)`"]
    end
    subgraph GitHub
    nilearn_GH["`nilearn
    (build)`"]
    nilearn_doc["`nilearn.github.io
    (host)`"]
    end
    nilearn_GH --> nilearn_doc
```

**Comment by Remi-Gau** [comment]
I will merge this one so I can confirm this works "in production".

**Pull request merged**